stremio-core-protobuf/src/model/fields/catalogs_with_extra.rs

```diff
@@ -17,33 +17,53 @@ impl FromProtobuf<Selected> for models::catalogs_with_extra::Selected {
 }
 
 impl ToProtobuf<models::catalogs_with_extra::Selected, ()> for Selected {
-    fn to_protobuf(&self, _args: &()) -> models::catalogs_with_extra::Selected {
+    fn to_protobuf<E: stremio_core::runtime::Env + 'static>(
+        &self,
+        _args: &(),
+    ) -> models::catalogs_with_extra::Selected {
         models::catalogs_with_extra::Selected {
             r#type: self.r#type.clone(),
-            extra: self.extra.to_protobuf(&()),
+            extra: self.extra.to_protobuf::<E>(&()),
         }
     }
 }
 
 impl ToProtobuf<models::Catalog, Ctx> for Catalog<MetaItemPreview> {
-    fn to_protobuf(&self, ctx: &Ctx) -> models::Catalog {
+    fn to_protobuf<E: stremio_core::runtime::Env + 'static>(&self, ctx: &Ctx) -> models::Catalog {
         models::Catalog {
-            pages: self.iter().map(|page| page.to_protobuf(ctx)).collect(),
+            pages: self.iter().map(|page| page.to_protobuf::<E>(ctx)).collect(),
         }
     }
 }
 
 impl ToProtobuf<models::CatalogsWithExtra, Ctx> for CatalogsWithExtra {
-    fn to_protobuf(&self, ctx: &Ctx) -> models::CatalogsWithExtra {
+    fn to_protobuf<E: stremio_core::runtime::Env + 'static>(
+        &self,
+        ctx: &Ctx,
+    ) -> models::CatalogsWithExtra {
         models::CatalogsWithExtra {
-            selected: self.selected.to_protobuf(&()),
-            catalogs: self.catalogs.to_protobuf(ctx),
+            selected: self.selected.to_protobuf::<E>(&()),
+            catalogs: self.catalogs.to_protobuf::<E>(ctx),
         }
     }
 }
 
+#[cfg(feature = "model-deser")]
+impl stremio_core_web::model::SerializeModel<Vec<u8>> for models::CatalogsWithExtra {
```

Comment:
IMO there shouldn't be any references to `stremio_core_web` as it's in no way related. Prob this means that this package is named incorrectly, or is misused/misplaced.

Reply:
Yes this is correct. The current stremio-core-web has a lot of the logic that we need for the UI but it's inaccessible to us as it's for wasm. I was playing with the idea of combining the core-web crate as dependency.
My goal is to introduce something like "stremio-core-ui" which all the core-* can share and the logic for the UI (like de-duplication for example) can be put there.

I will remove this impl for now as it was just an example I created.